src/components/RouteGuard.tsx

```diff
@@ -0,0 +1,66 @@
+import LoginModal from "@/components/LoginModal";
+import { useEffect, useState } from "react";
+
+const RouteGuard = ({ children }: Readonly<{ children: React.ReactNode }>) => {
+  const [isAuth, setAuth] = useState(false);
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const handleLogin = async (username: string, password: string) => {
+    const headers = new Headers();
+    headers.append("Authorization", "Basic " + btoa(username + ":" + password));
+    const res = await fetch(process.env.API_URL + "/login", {
```

Comment:
By default, let's use the internal nextjs API. One advantage of that is that we keep secrets and sensitive stuff on server side.

Reply:
The server api for login and signup has been created and is used throughout the login/signup process